### PR TITLE
Add requirement on using best practices

### PR DESCRIPTION
**Pull request by lehors**
This adds a small section indicating that all implementations MUST use industry security best practices.

Signed-off-by: Arnaud J Le Hors <lehors@us.ibm.com>

### CODE REVIEW

**Comment by lehors** [comment]
My goal is to address the gaping hole we currently have in the spec which fails to define a baseline for what implementations are required to do to ensure basic security and be able to claim compliance.
I propose to use the CIS controls as a non-prescriptive reference while recognizing that many other references could be used instead. Other people, like @david-a-wheeler , may have other ideas.

File: docs/spec/v1.0/requirements.md [hunk 1]
```diff
@@ -75,6 +75,17 @@ Responsibility to implement SLSA is spread across the following parties.
...
+### Security Best Practices
+
+While the exact definition of what constitutes a secure system is beyond the
+scope of this specification, to be conformant all implementations MUST use
+industry security best practices. This includes, but is not limited to, using
+proper access controls, securing communications, implementing proper management
+of cryptographic secrets, doing frequent updates, and promptly fixing known
+vulnerabilities.
+
+Various relevant standards and guides can be consulted for that matter such as the [CIS Critical Security Controls](https://www.cisecurity.org/controls/cis-controls-list).
```
Comment (MarkLodato) [inline docs/spec/v1.0/requirements.md:87]: nit: wrap at 80 columns for consistency

**Review by MarkLodato — APPROVED** [review]
Thanks!

File: docs/spec/v1.0/requirements.md (hunk 1 above)
Comment (lehors) [reply docs/spec/v1.0/requirements.md]: Sorry, I had missed that one. Fixed.

**Review by joshuagl — APPROVED** [review]

**Review by kpk47 — APPROVED** [review]

**Pull request merged**